### PR TITLE
Test conduct code snippet

### PR DESCRIPTION
**Pull request by asif-khan17**
**Fixed Issues**

#23100

**Files Changed**

test.component.ts
test.html
TestRepository.cs
TestConductRepositoryTest.cs
TestConductRepository.cs

**Checks**

- [X] Naming Conventions
- [X] Server side code comments (proper English, grammar and no spelling mistakes)
- [X] Client side code comments (proper English, grammar and no spelling mistakes)
- [X] Unused variables, methods, blank spaces and name spaces. 
- [X] Optimal Code and code formatting
- [X] Commit History is proper, no merge is done. 
- [X] Commit/pull request messages should be proper to justify your feature
- [ ] All test cases are executed provided by tester.

**Known Issues**

- issue 1
- issue 2

**Comments**

NA

---
This change is [Reviewable](https://reviewable.io/reviews/promact/trappist/393)

### CODE REVIEW

File: Trappist/src/Promact.Trappist.Repository/TestConduct/TestConductRepository.cs [hunk 1]
```diff
@@ -93,8 +93,9 @@ public TestConductRepository(TrappistDbContext dbContext, ITestsRepository testR
             var currentDate = DateTime.UtcNow;
             await _dbContext.TestIpAddresses.Where(x => x.TestId == testObject.Id).ToListAsync();
             // if Test is not paused and current date is not greater than EndDate and machine IP address is in the list of test ip addresses of  then it returns true and test link exist otherwise link does not exist
-           
-                return testObject != null && DateTime.Compare(currentDate, testObject.EndDate) < 0 && !testObject.IsPaused && testObject.TestIpAddress != null && testObject.TestIpAddress.Any(x => x.IpAddress == userIp) ? true : false;
+            if (testObject.TestIpAddress != null)
+                return testObject != null && DateTime.Compare(currentDate, testObject.EndDate) < 0 && !testObject.IsPaused && testObject.TestIpAddress.Any(x => x.IpAddress == userIp) ? true : false;
+            else return testObject != null && DateTime.Compare(currentDate, testObject.EndDate) < 0 && !testObject.IsPaused ? true : false;
```
Comment (chintans) [inline Trappist/src/Promact.Trappist.Repository/TestConduct/TestConductRepository.cs]: MINOR Remove the unnecessary Boolean literal(s). [rule](http://server12:9005/coding_rules#rule_key=csharpsquid%3AS1125)

**Comment by chintans** [comment]
SonarQube analysis reported 1 issue
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: Trappist/src/Promact.Trappist.Repository/Test/TestsRepository.cs [hunk 2]
```diff
@@ -324,7 +324,7 @@ await _dbContext.TestQuestion
                     question.CodeSnippetQuestion = Mapper.Map<CodeSnippetQuestionAC>(x.Question.CodeSnippetQuestion);
                     question.SingleMultipleAnswerQuestion = Mapper.Map<SingleMultipleAnswerQuestionAC>(x.Question.SingleMultipleAnswerQuestion);
                     //Removing correct answer(s)
-                    question.SingleMultipleAnswerQuestion.SingleMultipleAnswerQuestionOption.ForEach(y => y.IsAnswer = false);
+                    question.SingleMultipleAnswerQuestion?.SingleMultipleAnswerQuestionOption.ForEach(y => y.IsAnswer = false);
```
Comment (rushi216) [inline Trappist/src/Promact.Trappist.Repository/Test/TestsRepository.cs]: why ?

Comment (asif-khan17) [inline Trappist/src/Promact.Trappist.Repository/Test/TestsRepository.cs]: for the time being so that it can take null value also

Comment (rushi216) [inline Trappist/src/Promact.Trappist.Repository/Test/TestsRepository.cs]: we use ? operator only when assigning it to other variable or in if statement, this is not the proper way

**Pull request merged**